src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -117,17 +114,16 @@ public CachingContext(JsonSerializerOptions options)
                     return typeInfo;
                 }
 
-                typeInfo = Options.GetTypeInfoInternal(type);
+                typeInfo = Options.GetTypeInfoNoCaching(type);
                 if (typeInfo != null)
                 {
-                    return _jsonTypeInfoCache.GetOrAdd(type, _ => typeInfo);
+                    return _jsonTypeInfoCache.GetOrAdd(type, typeInfo);
                 }
 
                 return null;
```

Comment:
we can simplify this method now and just cache nulls. Previously null meant we had to check again because we could have added fallback for JsonSerializerContext but now we're not doing that anymore and null means we always throw anyway (if not liking nulls consider caching dummy JsonTypeInfo instance which is already configured and has Exception set so it always throws on usage although null sounds completely fine here)

Reply:
Why not? The cache calls into `GetTypeInfoNoCaching` which must account for `_typeInfoResolver` being null so I don't see why the caching context shouldn't. The only motivation might be that are able to cache null results but I don't think there are benefits in doing that. We're not really caching any results for which the resolver is throwing exceptions so I don't see why we should be doing it for null which are likely more rare.

Reply:
null means default resolver so it's effectively not null